### ISSUE TITLE
Some question about the code 

### ISSUE THREAD

**Issue by taluos** [hamzapehlivan/StyleRes#6]
Dear author, I can't find where the code for these two modules is as shown in the image. Can you tell me where i can find them, thank you.

<img width="226" alt="屏幕截图 2024-05-10 154907" src="https://github.com/hamzapehlivan/StyleRes/assets/67951309/f332801f-cef7-481d-a564-0fa646fc3e16">

**Comment by hamzapehlivan** [comment]
Hello, they are embedded into StyleGAN generator. 
This if statement is the starting point of forward pass for them: https://github.com/hamzapehlivan/StyleRes/blob/9c62ab7497206d47346148d4090fcf5e92e7e199/models/stylegan2.py#L583

**Comment by taluos** [comment]
Hello, I appreciate your feedback and thank you for the training code you sent me via email. After further training, I have some additional questions：

1：After 500,000 iterations of training the model, I achieved an FID score of 11.4 on CelebA, which is good progress, but still quite far from the 7 reported in the paper. My question is: approximately how many iterations are needed to achieve similar results as in the paper?

2：If I want to conduct ablation experiments, such as removing E1 or E2 individually, how should I proceed?

**Comment by shartoo** [comment]
Can this repo be trained on a single GPU(12GB) ?

**Comment by taluos** [comment]
> Can this repo be trained on a single GPU(12GB) ?

Absolutely, I have used a P4 (8GB) for model training. However, the training speed might be quite slow.

**Comment by shartoo** [comment]
> > Can this repo be trained on a single GPU(12GB) ?
> 
> Absolutely, I have used a P4 (8GB) for model training. However, the training speed might be quite slow.

How long does it take to train a new model,i need a 512x512 model with some new dataset.

**Comment by taluos** [comment]
> > > Can this repo be trained on a single GPU(12GB) ?
> > 
> > 
> > Absolutely, I have used a P4 (8GB) for model training. However, the training speed might be quite slow.
> 
> How long does it take to train a new model,i need a 512x512 model with some new dataset.

I haven't run a complete training session using a similar GPU, but I can give you some references: currently, I use three RTX 3090 GPUs (24GB) to iterate 300,000 rounds, which takes 20 hours. I think using a single 12GB GPU to iterate 100,000 rounds might take 40 hours (time required for one iteration is less than 1 second).

**Comment by shartoo** [comment]
Thank you for your sharing! The training process looks good,i need to train a 512x512 model, which should be 4 times shorter time to train.I'll try and feedback!

**Comment by shartoo** [comment]
Can you share the email address of author of this repo? the address "hamza.pehlivan@bilkent.edu.tr " from the paper seems invalid.

**Comment by taluos** [comment]
> Can you share the email address of author of this repo? the address "[hamza.pehlivan@bilkent.edu.tr](mailto:hamza.pehlivan@bilkent.edu.tr) " from the paper seems invalid.
This is the correct email address. You might need to wait for a while until the author responds (it may take up to a week).

**Comment by taluos** [comment]
> Can you share the email address of author of this repo? the address "[hamza.pehlivan@bilkent.edu.tr](mailto:hamza.pehlivan@bilkent.edu.tr) " from the paper seems invalid.

This is the correct email address. You might need to wait for a while until the author responds (it may take up to a week).

**Comment by shartoo** [comment]
i'm sending  with gmail, it shows invalid email address
![230](https://github.com/hamzapehlivan/StyleRes/assets/5716282/0f370c55-593c-468e-a60e-05c571ed067c)

**Comment by taluos** [comment]
> i'm sending with gmail, it shows invalid email address ![230](https://private-user-images.githubusercontent.com/5716282/347325667-0f370c55-593c-468e-a60e-05c571ed067c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA2MDEyMDgsIm5iZiI6MTcyMDYwMDkwOCwicGF0aCI6Ii81NzE2MjgyLzM0NzMyNTY2Ny0wZjM3MGM1NS01OTNjLTQ2OGUtYTYwZS0wNWM1NzFlZDA2N2MucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcxMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MTBUMDg0MTQ4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzY5MzIyYWU2YzEwYTFhZTkyMmFmZjI0NmE0MTA5MjgxZTY3MTBkY2E0OWMxZjExNzVlNTNjMTllZWYxZTg3YSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.Phs89QLMxkp7NRdar_GD3Pvd_nMh-MXN2X0a62GzSt4)

I used my QQ email to send the email.
![image](https://github.com/hamzapehlivan/StyleRes/assets/67951309/16bb3ab1-9e34-4f78-b6ed-a3816ed68d8e)
![image](https://github.com/hamzapehlivan/StyleRes/assets/67951309/4aa003b8-ae64-4a04-b639-d0d71d7b7550)

**Comment by shartoo** [comment]
Still failed to send even with QQ Mail
![18](https://github.com/hamzapehlivan/StyleRes/assets/5716282/f34b2568-711b-4c7f-a6de-9714febf71b9)

**Comment by shartoo** [comment]
My gmail is shartoo518@gmail.com, would you mind forwarding me a copy of the e-mail?

**Comment by hamzapehlivan** [comment]
@shartoo My previous email is closed, so I sent you from another one.

**Comment by hamzapehlivan** [comment]
> Hello, I appreciate your feedback and thank you for the training code you sent me via email. After further training, I have some additional questions：
> 
> 1：After 500,000 iterations of training the model, I achieved an FID score of 11.4 on CelebA, which is good progress, but still quite far from the 7 reported in the paper. My question is: approximately how many iterations are needed to achieve similar results as in the paper?
> 
> 2：If I want to conduct ablation experiments, such as removing E1 or E2 individually, how should I proceed?

1-) FID is a very sensitive metric to resizing method, image format (JPG/PNG) and other things as well. Please see this [link ](https://github.com/GaParmar/clean-fid) for detailed explanation. 

In this repo, we also shared the exact code to reproduce the FID results. After you save the model coming from the training code, load it from this repo, and save all the images (with resizing, they should be 256x256, we have a --resize option in this repo). Then, run the provided fid.sh file.

2-) You should remove either E1 or E2 in the training code and let it run. Unfortunately, there is no flag to control which encoders are used. 

I think you already trained the model, but there might be problems with FID calculations.

**Comment by shartoo** [comment]
Some pretrained model is missing from the training code, i dont know if i found the correct source
+ vgg16.pt:  https://download.pytorch.org/models/vgg16-397923af.pth  ,found from https://github.com/minar09/VGG16-PyTorch/blob/master/vgg.py
+ arcface.pth:  https://github.com/ronghuaiyang/arcface-pytorch

code from `styleres_cfg.py`

```
gan_model_path = 'checkpoints/stylegan2_official_ffhq1024.pth'
discriminator_model_path = 'checkpoints/stylegan2_official_ffhq256.pth'
perceptual_model_path = 'checkpoints/vgg16.pth'
arcface_path = 'checkpoints/arcface.pth'
```

**Comment by hamzapehlivan** [comment]
We used the same VGG network used in [genforce](https://github.com/genforce/genforce/tree/master).  Here is the [link](https://drive.google.com/file/d/1qQ-r7MYZ8ZcjQQFe17eQfJbOAuE3eS0y/view?usp=sharing) for the weights. 

Here is the [arcface](https://drive.google.com/file/d/1KW7bjndL3QG3sxBbZxreGHigcCCpsDgn/view). It is also used in HyperStyle and pSp.

**Comment by hamzapehlivan** [comment]
> > Hello, I appreciate your feedback and thank you for the training code you sent me via email. After further training, I have some additional questions：
> > 1：After 500,000 iterations of training the model, I achieved an FID score of 11.4 on CelebA, which is good progress, but still quite far from the 7 reported in the paper. My question is: approximately how many iterations are needed to achieve similar results as in the paper?
> > 2：If I want to conduct ablation experiments, such as removing E1 or E2 individually, how should I proceed?
> 
> 1-) FID is a very sensitive metric to resizing method, image format (JPG/PNG) and other things as well. Please see this [link ](https://github.com/GaParmar/clean-fid) for detailed explanation.
> 
> In this repo, we also shared the exact code to reproduce the FID results. After you save the model coming from the training code, load it from this repo, and save all the images (with resizing, they should be 256x256, we have a --resize option in this repo). Then, run the provided fid.sh file.
> 
> 2-) You should remove either E1 or E2 in the training code and let it run. Unfortunately, there is no flag to control which encoders are used.
> 
> I think you already trained the model, but there might be problems with FID calculations.

One more thing to consider is, you do not have to follow our FID calculation directly. If you apply the same preprocessing/FID network etc. to all methods, their relative order should not change. 

If your model performs better than the all other methods with your FID implementation, it will probably be the best with our implementation as well.

**Comment by phuongpropk05** [comment]
dear author, I dont understand the name field for each column in ganspace_config.csv file, can you give me the detail for it. And can you give me more parameter of others styles. Thank you. 
![Screenshot from 2024-07-17 16-44-26](https://github.com/user-attachments/assets/4bc7c25d-3b79-4292-b2a6-cb990a6ccf98)

**Comment by shartoo** [comment]
The detail of [GANSpace](https://github.com/harskish/ganspace)

**Comment by phuongpropk05** [comment]
@shartoo can you forward me the training code beacause I dont know the email of the author

**Comment by phuongpropk05** [comment]
@shartoo can you forward me the training code beacause I dont know the email of the author. My email is hoangphuong070501@gmail.com. Thank you!

**Comment by hkchae96** [comment]
@hamzapehlivan @shartoo Could I also request the training code? I had the same problem with your invalid email address. My email is hkchae96@gmail.com. Thank you.

**Comment by shartoo** [comment]
> @hamzapehlivan @shartoo Could I also request the training code? I had the same problem with your invalid email address. My email is [hkchae96@gmail.com](mailto:hkchae96@gmail.com). Thank you.

Better ask the author to send the code, I'm not authorized to send

**Comment by taluos** [comment]
@hamzapehlivan 
I'm trying to use test.sh to generate images, but I found that I don't have the edit_configs.py file required by test.py. This file wasn't used during the training process either. Could you provide that file to me?
`
    parser.add_argument('--edit_cfg', type=str, default='configs/edit_configs.py',
                        help='Launcher type.')
`
